src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -271,14 +272,17 @@ public void setApplicationEventPublisher(ApplicationEventPublisher applicationEv
 
     @Override
     @ExternalAuthorization(resource = Resource.CERTIFICATE, action = ResourceAction.LIST, parentResource = Resource.RA_PROFILE, parentAction = ResourceAction.MEMBERS)
-    public CertificateResponseDto listCertificates(SecurityFilter filter, SearchRequestDto request) {
+    public CertificateResponseDto listCertificates(SecurityFilter filter, CertificateSearchRequestDto request) {
         setupSecurityFilter(filter);
         RequestValidatorHelper.revalidateSearchRequestDto(request);
         final Pageable p = PageRequest.of(request.getPageNumber() - 1, request.getItemsPerPage());
-
-        final TriFunction<Root<Certificate>, CriteriaBuilder, CriteriaQuery, Predicate> additionalWhereClause = (root, cb, cr) -> FilterPredicatesBuilder.getFiltersPredicate(cb, cr, root, request.getFilters());
+        List<SearchFilterRequestDto> filters = new ArrayList<>(request.getFilters());
+        if (!request.isIncludeArchived()) {
```

Comment:
This should be handled by adding predicate for that field to additionalWhereClause through `TriFunction`.